File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 12/6/2025, 5:55:52 am
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ret;
        for (int i=0; i<nums.size(); i++){
            for (int j=i+1; j<nums.size(); j++){
                if (nums[i]+nums[j] == target){
                    ret.push_back(i);
                    ret.push_back(j);
                    return ret;
                }
            }
        }return ret;

    }
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 12/6/2025, 5:55:49 am
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}
        for i in range(len(nums)):
            if nums[i] in hash: return [i, hash[nums[i]]]
            else:
                hash[target-nums[i]] = i
        return []